parts/linux/cloud-init/artifacts/components.json

```diff
@@ -870,6 +1286,11 @@
       "name": "nvidia-container-toolkit",
       "downloadLocation": "/usr/local/bin",
       "downloadURIs": {
+        "windows": {
```

Comment:
Can you also make sure the script will not crash if the "windows" block isn't presented? Please ignore if it's already handled.
Not sure about Windows node, but in Linux case, there could be cases like new shell scripts are executing in old VHD image (security patch scenarior) which doesn't always have the latest components.json thus some keys could be missing. We encountered some IcMs because of this.

Reply:
It's already handled, but I have added some powershell unit tests for this case. 

However, Windows doesn't parse components.json at provision time - it's only at build time - so the risk of IcM is less for Windows than linux anyway.